edx_solutions_api_integration/mobile_api/views.py

```diff
@@ -153,25 +160,18 @@ def _get_user_course_grades(self, user, course_id):
         Once persistent grades are enabled on the solutions fork, we'll use CourseGradeFactory instead.
         """
         course_key = get_course_key(course_id)
-        try:
-            record = StudentGradebook.objects.get(user=user, course_id=course_key)
-            course_grade = record.grade
-        except StudentGradebook.DoesNotExist:
-            course_grade = 0
+        data = get_cached_data('grade', course_id, user.id)
+        params = {'exclude_users': get_aggregate_exclusion_user_ids(course_key, roles=None)}
 
-        course_average = self._get_course_average_grade(course_key)
-        return {
-            'course_grade': course_grade,
-            'course_average_grade': course_average
-        }
+        if not data:
+            course_avg = StudentGradebook.course_grade_avg(course_key, **params)
+            user_grade = StudentGradebook.get_user_grade(course_key, user.id)
 
-    def _get_course_average_grade(self, course_key):
-        """
-        Get the average grade for all the users in the specified course.
+            data = {'user_grade': user_grade, 'course_avg': course_avg}
+            cache_course_data('grade', course_id, {'course_avg': course_avg})
+            cache_course_user_data('grade', course_id, user.id, {'user_grade': user_grade})
```

Comment:
does the user grade cache expires when user course grade is updated? because user would expect his own grade updated immediately, course average can be cached.

Reply:
Yes it does. https://github.com/edx-solutions/gradebook-edx-platform-extensions/blob/master/gradebook/signals.py#L80